skfs/src/Context.sk

```diff
@@ -1626,4 +1626,52 @@ fun withRegionValue<T>(f: () ~> T): T {
   copied.getHead();
 }
 
+fun withRegionFold<K, T>(
+  contextOpt: ?mutable Context,
+  it: mutable Iterator<K>,
+  value: T,
+  fn: (?mutable Context, K, T) ~> T,
+): T {
+  optSaved: ?Obstack = None();
+  valueInRegion = value;
+  loop {
+    if (
+      try {
```

Comment:
Am I confused, or could the whole `loop` be in the body of the `try`? I'm not entirely sure of the compilation scheme, but doing that might save some exception setup and cleanup on each loop iteration.